### ISSUE TITLE
Aromatic molecule generated from adjlist does not produced kekulized version in resonance isomers

### ISSUE THREAD

**Issue by connie** [ReactionMechanismGenerator/RMG-Py#405]
I thought we do the kekulization at the start but this is problematic for importing from Java.

Loading a molecule from SMILES produces the two versions of isomers (both aromatic and kekulized).

    >>> m = Molecule().fromSMILES('Cc1ccccc1')
    >>> m.generateResonanceIsomers()
    [Molecule(SMILES="Cc1ccccc1"), Molecule(SMILES="Cc1ccccc1")]

Loading it from an aromatic adjlist retains only the aromatic version.  However, if you load from a kekulized adjlist you get both resonance isomers.

    >>> m2 = Molecule().fromAdjacencyList('''1  H 0 {2,S}
    ... 2  C 0 {3,S} {9,S} {10,S} {1,S}
    ... 3  C 0 {4,B} {8,B} {2,S}
    ... 4  C 0 {3,B} {5,B} {11,S}
    ... 5  C 0 {4,B} {6,B} {12,S}
    ... 6  C 0 {5,B} {7,B} {13,S}
    ... 7  C 0 {6,B} {8,B} {14,S}
    ... 8  C 0 {3,B} {7,B} {15,S}
    ... 9  H 0 {2,S}
    ... 10  H 0 {2,S}
    ... 11  H 0 {4,S}
    ... 12  H 0 {5,S}
    ... 13  H 0 {6,S}
    ... 14  H 0 {7,S}
    ... 15  H 0 {8,S}
    ... ''')

    >>> m2.generateResonanceIsomers()
    [Molecule(SMILES="Cc1ccccc1")]

**Comment by rwest** [comment]
I think the call `fromSMILES`, via RDKit, kekulizes it in RDKit.
We could add an "if aromatic then kekulize using RDKit" step in the `generateResonanceIsomers()` method.

I also notice that `toSMILES` seems to "unkekulize" it - in that both resonance isomers in your example above are printed as `Molecule(SMILES="Cc1ccccc1")` even though their adjacency lists are different. This may be unavoidable and hopefully is not a problem if we're mindful that SMILES are a sometimes-conevinent shorthand, and the "true" representation is always the adjacency list.

**Comment by connie** [comment]
Yeah it looks like SMILES always return the aromatic representation, so the shorthand is indeed deceiving.

Similarly on print issues, the other day I created a Species object with a single molecule:

    >>> from rmgpy.molecule import *
    >>> from rmgpy.species import *
    >>> m = Molecule().fromSMILES('c1ccccc1')
    >>> s = Species(molecule = [m])
    >>> s.generateResonanceIsomers()
    >>> s
    Species(label="", molecule=[Molecule(SMILES="c1ccccc1")])
    >>> s.molecule
    [Molecule(SMILES="c1ccccc1"), Molecule(SMILES="c1ccccc1")]

So basically the representation of s shows only 1 molecule, when in fact it actually contains 2.

**Comment by rwest** [comment]
That's really weird, and should be fixed

**Comment by rwest** [comment]
I noticed this in `Species.generateResonanceIsomers()`
```python
    def generateResonanceIsomers(self):
        """
        Generate all of the resonance isomers of this species. The isomers are
        stored as a list in the `molecule` attribute. If the length of
        `molecule` is already greater than one, it is assumed that all of the
        resonance isomers have already been generated.
        """
        if len(self.molecule) == 1:
            self.molecule = self.molecule[0].generateResonanceIsomers()
```
Why is that assumption *("If the length of `molecule` is already greater than one, it is assumed that all of the resonance isomers have already been generated.")* helpful?
